test/http_download_test.rb

```diff
@@ -49,3 +51,22 @@ def test_should_skip_download_if_one_is_in_progress
     end
   end
 end
+
+class HttpDownloadsIntegrationTest < Test::Unit::TestCase
+  include Proxy::IntegrationTestCase
```

Comment:
Thinking more about this: this launches a full Smart Proxy but really we need a (minimal) webserver. I'm tempted to only launch a minimal webrick server so curl can work. Then we can also add some more integration tests like verifying that HTTP 200 works.